test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,402 @@
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+			gomega.Eventually(func() bool {
+				err := k8sClient.Get(ctx, lookupKey, createdPod)
+				return err == nil && !slices.Contains(createdPod.Finalizers, "kueue.x-k8s.io/managed")
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue(),
+				"Expected 'kueue.x-k8s.io/managed' finalizer to be removed")
+		})
+
+		ginkgo.It("Should stop the single pod with the queue name if workload is evicted", func() {
+			ginkgo.By("Creating a pod with queue name")
+			pod := podtesting.MakePod(podName, ns.Name).Queue("test-queue").Obj()
+			gomega.Expect(k8sClient.Create(ctx, pod)).Should(gomega.Succeed())
+			lookupKey := types.NamespacedName{Name: podName, Namespace: ns.Name}
+			createdPod := &corev1.Pod{}
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, lookupKey, createdPod)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			gomega.Expect(hasKueueSchedulingGate(createdPod)).To(gomega.BeTrue(),
+				"Pod should have 'kueue.x-k8s.io/admission' scheduling gate set")
+
+			gomega.Expect(hasManagedLabel(createdPod)).To(gomega.BeTrue(),
+				"Pod should have 'kueue.x-k8s.io/managed=true' label set")
+
+			gomega.Expect(createdPod.Finalizers).To(gomega.ContainElement("kueue.x-k8s.io/managed"),
+				"Pod should have 'kueue.x-k8s.io/managed' finalizer set")
+
+			ginkgo.By("checking that workload is created for pod with the queue name")
+			createdWorkload := &kueue.Workload{}
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			gomega.Expect(createdWorkload.Spec.PodSets).To(gomega.HaveLen(1))
+
+			gomega.Expect(createdWorkload.Spec.QueueName).To(gomega.Equal("test-queue"), "The Workload should have .spec.queueName set")
+
+			ginkgo.By("checking that pod is unsuspended when workload is assigned")
+			clusterQueue := testing.MakeClusterQueue("cluster-queue").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "1").Obj(),
+				).Obj()
+			admission := testing.MakeAdmission(clusterQueue.Name).
+				Assignment(corev1.ResourceCPU, "default", "1").
+				AssignmentPodCount(createdWorkload.Spec.PodSets[0].Count).
+				Obj()
+			gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+			util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdPod); err != nil {
+					return false
+				}
+				return !hasKueueSchedulingGate(createdPod)
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Eventually(func() bool {
+				ok, _ := testing.CheckLatestEvent(ctx, k8sClient, "Started", corev1.EventTypeNormal, fmt.Sprintf("Admitted by clusterQueue %v", clusterQueue.Name))
```

Comment:
Maybe for a follow up, but there are two alternatives here to better implement this, and have more comprehensive error messages:
- Have `CheckLatestEvent` only return an error and use `gomega.Succeed()`
- Implement a gomega matcher.

Reply:
I prefer `Implement a gomega matcher.`.

Reply:
I went with the first option, since there's not much business logic needed for a custom gomega matcher. We will have to fetch all the events before running a matcher anyway.

See: 

https://github.com/kubernetes-sigs/kueue/blob/1c7a932045462bf96865a33213d228549dd02634/test/integration/controller/jobs/pod/pod_controller_test.go#L155-L159